addons/sales_team/models/crm_team.py

```diff
@@ -44,7 +44,7 @@ def _get_default_team_id(self, user_id=None, domain=None):
         :param user_id: salesperson to target, fallback on env.uid;
         :domain: optional domain to filter teams (like use_lead = True);
         """
-        if user_id is None:
+        if not user_id:
```

Comment:
We could even say default value is False instead of None in addition to your fix. That way it matches Odoo default values when an ID is not given.

Reply:
updated